.github/workflows/main.yml

```diff
@@ -128,15 +128,61 @@ jobs:
 
   publish-flextesa-plugin-to-npm:
     needs:
-      - publish-ligo-plugin-to-npm
-      - publish-smartpy-plugin-to-npm
-      - publish-taquito-plugin-to-npm
+      - publish-sdk-to-npm
     uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@main
     with:
       dir: taqueria-plugin-flextesa
     secrets:
       NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
 
+  run-tests:
+    needs: build-binaries
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./tests
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16.x
+
+      - uses: denoland/setup-deno@v1
+        with:
+          deno-version: "^1.0.0"
+
+      - name: get binary and make executable from previous step for tests
+        run: |
+          wget ${{ needs.build-binaries.outputs.public-url-ubuntu }} -O taq
+          chmod +x taq
+          echo "/home/runner/.deno/bin" >> $GITHUB_PATH
+          echo "$(pwd)" >> $GITHUB_PATH
+          echo "$PATH"
+
+      - name: install dependencies for e2e tests in root and test directories
+        run: |
+          cd ..
+          echo "$PATH"
+          npm ci
+          npm install --workspace=taqueria-plugin-ligo
+          npm install --workspace=taqueria-plugin-mock
+          npm install --workspace=tests
+      
+      - name: Run unit tests
+        run: |
+          export PATH="/home/runner/.deno/bin:$PATH"
+          npm run test:unit
+
+      - name: Run integration tests
+        run: |
+          export PATH="$(pwd):$PATH"
+          npm run test:integration
+
+      - name: Run e2e tests
+        run: |
+          export PATH="$(pwd):$PATH"
+          npm run test:e2e 
```

Comment:
Once you remove the default `working-directory` you'll need to run these commands from root folder with `npm run test:e2e --workspace test`.